.github/workflows/autobuild.yaml

```diff
@@ -1,41 +1,125 @@
-# Workflow to build binaries for a release.
-# Triggered by release creation, which should include `llvm-project@<full-sha>`.
+# Workflow to build binaries and release them.
+# Triggered by the schedule or manual dispatch, which might include
+# `<owner/llvm-project>@<full-sha>`.
 #
 # Because the build takes more than an hour, our GITHUB_TOKEN credentials may
 # expire. A token `secrets.RELEASE_TOKEN` must exist with public_repo scope.
 name: Build release binaries
 on:
-  release:
-    types: created
+  # Run weekly on sunday at 21:37 UTC (arbitrary)
+  schedule:
+  - cron: '37 21 * * SUN'
+  workflow_dispatch:
+    inputs:
+      commit:
+        description: 'Commit to build from'
+        required: false
+      repo:
+        description: 'Repository to build from'
+        required: false
+        default: 'llvm/llvm-project'
+      tag:
+        description: 'Tag name'
+        required: false
+        default: ''
 jobs:
-  prepare:
+  # Choose the commit to build a release from.
+  #
+  # We want to avoid unbuildable revisions: choose the last green from CI.
+  # FIXME: The criteria should be some consistent set of buildbots passing.
+  pick:
```

Comment:
nit: it might be better to have two separate jobs, one for `create_env_from_inputs` other for `create_env_from_head` and having single `if` conditions rather than a bunch of ifs scattered across.

i am not sure how the `needs` statements in the create_release step will look like though, as one of those actions will always fail. i've seen https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-not-requiring-dependent-jobs-to-be-successful but not sure if it applies to our case.

Reply:
Replying to Sam for a similar comment.